### PR TITLE
feat(gcp): add a `test_connection` method

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Context

We need to be able to test the connection to each provider using one of the available authentication methods, starting from the environment variables.

### Description
Add a `test_connection` method for the GCP provider.
Create a `test_connection` abstract method in the Provider class but not enforced for now not to break anything.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/gcp/gcp_provider.py [hunk 1]
```diff
@@ -238,12 +243,46 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
...
 
-    def __set_gcp_creds_env_var__(self, credentials_file):
-        logger.info(
-            "GCP provider: Setting GOOGLE_APPLICATION_CREDENTIALS environment variable..."
-        )
-        client_secrets_path = os.path.abspath(credentials_file)
-        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = client_secrets_path
+    @staticmethod
+    def test_connection(
+        credentials_file: str, service_account: str, raise_on_exception: bool = True
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py]: Is it mandatory to set the `credentials_file` or `service_account`? If not please set some default values.

File: prowler/providers/gcp/gcp_provider.py [hunk 2]
```diff
@@ -238,12 +243,46 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
...
+    ) -> Connection:
+        """
+        Test the connection with the provided credentials file or service account to impersonate
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py]: Improve the docstring with the possible exceptions raised, what the method do and more about the args and returns please.

File: prowler/providers/gcp/gcp_provider.py [hunk 3]
```diff
@@ -238,12 +243,46 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
...
+            service = discovery.build("cloudresourcemanager", "v1", credentials=session)
+            request = service.projects().list()
+            request.execute()
+            return Connection(is_connected=True)
+        except HttpError as http_error:
+            if "Cloud Resource Manager API has not been used" in str(http_error):
+                logger.critical(
+                    "Cloud Resource Manager API has not been used before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudresourcemanager.googleapis.com/ then retry."
+                )
+                raise Exception(
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py]: Remember to add `if raise_on_exception`.

File: tests/providers/gcp/gcp_provider_test.py
Comment (jfagoagas) [inline tests/providers/gcp/gcp_provider_test.py:1]: Please add tests for the `test_connection` method.

**Review by jfagoagas — DISMISSED** [review]

File: prowler/providers/gcp/gcp_provider.py [hunk 4]
```diff
@@ -238,12 +243,52 @@ def setup_session(self, credentials_file: str, service_account: str) -> Credenti
...
-        )
-        client_secrets_path = os.path.abspath(credentials_file)
-        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = client_secrets_path
+    @staticmethod
+    def test_connection(
+        credentials_file: str = None,
+        service_account: str = None,
+        raise_on_exception: bool = True,
+    ) -> Connection:
+        """
```
Comment (pedrooot) [inline prowler/providers/gcp/gcp_provider.py:252]: Could you add docstrings following the guide from https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings please? Thanks!

**Review by pedrooot — CHANGES_REQUESTED** [review]

**Review by pedrooot — APPROVED** [review]
🔝 Thanks for this amazing feature!!

**Pull request merged**